src/Microsoft.Performance.SDK/Processing/ColumnBuilding/ColumnVariantIdentifier.cs

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Microsoft.Performance.SDK.Processing.ColumnBuilding;
+
+/// <summary>
+///     Represents a identifier for a column variant.
+/// </summary>
+/// <param name="Guid">
+///     The unique identifier for the column variant.
+/// </param>
+/// <param name="Name">
+///     The human-readable name of the column variant.
+/// </param>
+public record ColumnVariantIdentifier(Guid Guid, string Name);
```

Comment:
>Guid Guid [](http://example.com/codeflow?start=38&length=9)

Do we have any checks on uniqueness? Do we want any? Seems like it might be helpful to catch accidental copy/paste errors early as that might be a pain to debug.

Reply:
I can add some checks. There are two options
1. Verify GUIDs as they are added with each method call
2. Check all GUIDs during the root's `Commit` (`Build`) call

Option 1 would provide more immediate feedback, but would require a walk of the entire data structure for every method call. It's also harder to implement.

Option 2 would be easier to implement, but provide less immediate feedback.

Reply:
Added option 2 (with a unit test)